src/sentry/lang/native/appconnect.py

```diff
@@ -0,0 +1,237 @@
+
+    :raises KeyError: if the config is not found.
+    """
+    sources_config = project.get_option(SYMBOL_SOURCES_PROP_NAME)
+
+    sources = json.loads(sources_config)
+    for source in sources:
+        if source.get("type") == SYMBOL_SOURCE_TYPE_NAME and source.get("id") == config_id.lower():
+            return source
+    else:
+        raise KeyError(f"No {SYMBOL_SOURCE_TYPE_NAME} symbol source found with id {config_id}")
+
+
+class ITunesClient:
+    """A client for the legacy iTunes API.
+
+    Create this by calling :class:`AppConnectClient.itunes_client()`.
+
+    On creation this will contact iTunes and will fail if it does not have a valid iTunes
+    session.
+    """
+
+    def __init__(self, itunes_cookie: str, itunes_org: int):
+        self._session = requests.Session()
+        itunes_connect.load_session_cookie(self._session, itunes_cookie)
+        # itunes_connect.set_provider(self._session, itunes_org)
+
+    def download_dsyms(self, build: BuildInfo, path: str) -> None:
+        # TODO(flub): is there a better type for the path?
+        url = itunes_connect.get_dsym_url(
+            self._session, build.app_id, build.version, build.build_number, build.platform
+        )
+        if not url:
+            raise NoDsymsError
+        logger.debug("Fetching dSYM from: %s", url)
+        with requests.get(url, stream=True) as req:
+            req.raise_for_status()
+            with open(path, "wb") as fp:
+                for chunk in req.iter_content(chunk_size=io.DEFAULT_BUFFER_SIZE):
+                    fp.write(chunk)
+
+
+class AppConnectClient:
+    """Client to interact with a single app from App Store Connect.
+
+    Note that on creating this instance it will already connect to iTunes to set the
+    provider for this session.  You also don't want to use the same iTunes cookie in
+    multiple connections, so only make one client for a project.
+    """
+
+    def __init__(
+        self,
+        api_credentials: appstore_connect.AppConnectCredentials,
+        itunes_cookie: str,
+        itunes_org: int,
+        app_id: str,
+    ) -> None:
+        """Internal init, use one of the classmethods instead."""
+        self._api_credentials = api_credentials
+        self._session = requests.Session()
+        self._itunes_cookie = itunes_cookie
+        self._itunes_org = itunes_org
+        self._app_id = app_id
+
+    @classmethod
+    def from_project(cls, project: Project, config_id: str) -> "AppConnectClient":
+        """Creates a new client for the project's appStoreConnect symbol source.
+
+        This will load the configuration from the symbol sources for the project if a symbol
+        source of the ``appStoreConnect`` type can be found which also has matching
+        ``credentials_id``.
+        """
+        config = get_app_store_config(project, config_id)
+        return cls.from_config(config)
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "AppConnectClient":
```

Comment:
I've seen `Config` type pop up a few times. Can we make this a type alias such that we at least know while reading that this is the same config? If the definition is consolidated it makes it easier to fill in with `TypedDict` at a later point too

Reply:
I've been wondering if I should be turning it into a dataclass as well really.  maybe i should bite the bullet and do that

Reply:
I mean `Config = Dict[str, Any]` would already help a lot for understanding